dashboard.py

```diff
@@ -1,603 +1,801 @@
-            tz=summary_context_timezone_change_type, 
-            freq=f"{day_offset}D", 
-            inclusive="right"
-        )
-
-    return [
-        dcc.RangeSlider(
-            id="summary_context_time_slider",
-            min=0, 
-            max=len(time_range) - 1, 
-            step=len(time_range),
-            value=[0, len(time_range) - 1],
-            marks={str(idx): time_range[idx].strftime('%m-%d') for idx in range(len(time_range))},
-            updatemode='drag'
-        )
-    ]
-
-if __name__ == '__main__': 
-    app.run_server()
+import dash
+from dash import html
+from dash import dcc
+from dash import dash_table
+import plotly.graph_objects as go
+import plotly.express as px
+from dash.dependencies import Input, Output
+import pandas as pd
+import sys
+import requests
+from config import TOKEN
+
+app = dash.Dash()
+
+
+MOOCLETS = {'315': 'datasets/Modular_Link_MHA_Prototype.xlsx',
+           '295': 'datasets/MHAwave1ModularRationale.xlsx'}
+
+# 315, 295:
+# MOOCLET_FILES = ['datasets/Modular_Link_MHA_Prototype.xlsx',
+#                  'datasets/MHAwave1ModularRationale.xlsx']
+# MOOCLET_URLS = ['https://mooclet.canadacentral.cloudapp.azure.com/engine/api/v1/policyparameters?mooclet=315&policy=6',
+#                 'https://mooclet.canadacentral.cloudapp.azure.com/engine/api/v1/policyparameters?mooclet=295&policy=6']
+# XLS = []
+# REWARD = ["modular_link_mha_prototype_linkrating",
+#           "MHAwave1ModularMessageReward"]
+# UR_DFS = []
+# TSC_DFS = []
+# DFS = []
+#
+# UR_DF = [0]
+# TSC_DF = [0]
+# DF = [0]
+# CONTEXTUAL = [[
+#     "modular_link_mha_prototype_hadrecentactivitylast48hours",
+#     "modular_link_mha_prototype_isweekend",
+#     "modular_link_mha_prototype_timeofday",
+#     "modular_link_mha_prototype_highmood",
+#     "modular_link_mha_prototype_highenergy",
+#     "modular_link_mha_prototype_studyday",
+#     "modular_link_mha_prototype_k10",
+#     "modular_link_mha_prototype_averageresponsiveness",
+#     "modular_link_mha_prototype_averagecontentratingsoverall"
+# ], [
+#     "wave1RecentActivityLast48Hours",
+#     "wave1IsWeekend",
+#     "wave1isHighMood",
+#     "wave1isHighEnergy",
+#     "wave1TimeOfDay",
+#     "wave1UserLevelRecencyRationale",
+#     "wave1PriorRationaleRating",
+#     "wave1StudyDay",
+#     "wave1K10",
+#     "wave1AverageResponsiveness",
+#     "wave1AverageContentRatingsOverall"
+# ]
+# ]
+
+MOOCLET_FILES = []
+MOOCLET_URLS = []
+XLS = []
+REWARD = []
+
+UR_DFS = []
+TSC_DFS = []
+DFS = []
+
+UR_DF = [0]
+TSC_DF = [0]
+DF = [0]
+
+CONTEXTUAL = []
+
+
+
+def upload_mooclets(file, num):
+    MOOCLET_FILES.append(file)
+    MOOCLET_URLS.append("https://mooclet.canadacentral.cloudapp.azure.com/engine/api/v1/policyparameters?mooclet={0}&policy=6".format(num))
+    XLS.append(pd.ExcelFile(file))
+
+
+def prepare():
+    for i in range(len(XLS)):
+        ur_df = pd.read_excel(XLS[i], 'UR_0')
+        ur_df[REWARD[i]] = ur_df[REWARD[i]] * 4 + 1

```

Comment:
TODO: Note that there are some rewards that are already scaled up in the dataset.

Reply:
OK the code is updated in the next commit.